File: leetcode_python/reverse_integer.py
class Solution:
    # @return an integer
    def reverse(self, x):
        x = list(str(x))
        initial_pos = 1 if x[0] == '-' else 0
        for i in range(initial_pos, (len(x) + initial_pos) / 2):
            x[i], x[-1 - i + initial_pos] = x[-1 - i + initial_pos], x[i]
        return int(''.join(x))

File: leetcode_cpp/reverse_integer.cpp
class Solution {
public:
    int reverse(int x) {
        string x_str = to_string(x);
        int initial_pos = (x_str[0] == '-') ? 1 : 0;
        char tmp;

        for (int i = initial_pos; i < (x_str.size() + initial_pos) / 2; ++i) {
            tmp = x_str[i];
            x_str[i] = x_str[x_str.size() + initial_pos - 1 - i];
            x_str[x_str.size() + initial_pos - 1 - i] = tmp;
        }

        return stoi(x_str);
    }
};
